src/store/ShareStore.js

```diff
@@ -4,6 +4,10 @@ import { downloadFileInBrowser } from '../utils';
 import Menu from '../components/Menu.jsx';
 import { logEvent, LogCategories, LogActions } from '../utils/ga';
 
+const is_firefox = navigator.userAgent.indexOf('Firefox') !== -1;
+const is_safari = navigator.userAgent.indexOf('Safari') !== -1 && navigator.userAgent.indexOf('Chrome') === -1;
+const is_iOS = !!navigator.userAgent.match(/iPhone|iPad|iPod/i);
```

Comment:
let's create a shared function for this. noticed that we check for browser in other components like `CurrentSpotStore.js`.

this is a good source: https://yogeshchauhan.com/javascript-how-to-detect-a-browser-using-the-user-agent/